deploy/modules/gcp/tidb-cluster/main.tf

```diff
@@ -0,0 +1,159 @@
+resource "google_container_node_pool" "pd_pool" {
+  // The monitor pool is where tiller must first be deployed to.
+  depends_on = [google_container_node_pool.monitor_pool]
+  provider   = google-beta
+  project    = var.gcp_project
+  cluster    = var.gke_cluster_name
+  location   = var.gke_cluster_location
+  name       = "${var.cluster_name}-pd-pool"
+  node_count = var.pd_node_count
+
+  management {
+    auto_repair  = false
+    auto_upgrade = false
+  }
+
+  node_config {
+    machine_type    = var.pd_instance_type
+    local_ssd_count = 0
+
+    taint {
+      effect = "NO_SCHEDULE"
+      key    = "dedicated"
+      value  = "${var.cluster_name}-pd"
+    }
+
+    labels = {
+      dedicated = "${var.cluster_name}-pd"
+    }
+
+    tags         = ["pd"]
+    oauth_scopes = ["storage-ro", "logging-write", "monitoring"]
+  }
+}
+
+resource "google_container_node_pool" "tikv_pool" {
```

Comment:
The tikv node pool is created first because it takes the longest due to the reasons listed by greg here https://github.com/pingcap/tidb-operator/pull/717#discussion_r310219898